web/server/codechecker_server/api/report_server.py

```diff
@@ -2511,6 +2511,9 @@ def get_analyzer_name(report):
             if report.metadata:
                 return report.metadata.get("analyzer", {}).get("name")
 
+            if report.check_name.startswith('clang-diagnostic-'):
+                return 'clang-tidy'
+
```

Comment:
Should we add some default case if non of the above return some analyzer name? Something like `unknown analyzer`?

Reply:
By default if this field is not set it will be 'unknown':
https://github.com/Ericsson/codechecker/blob/ff0b8c902bd99e68b6bebe18ccfb39ef60439e5d/web/server/codechecker_server/database/run_db_model.py#L305-L307